### PR TITLE
ArduCopter: implement user aux switch functions

### PR DESCRIPTION
**Pull request by SergeyBokhantsev**
Implementation for https://github.com/ArduPilot/ardupilot/issues/8493

### CODE REVIEW

**Comment by SergeyBokhantsev** [comment]
@rmackay9 could you please check if this is good to be merged?

File: ArduCopter/APM_Config.h [hunk 1]
```diff
@@ -65,3 +65,4 @@
 //#define USERHOOK_MEDIUMLOOP userhook_MediumLoop();        // for code to be run at 10hz
 //#define USERHOOK_SLOWLOOP userhook_SlowLoop();            // for code to be run at 3.3hz
 //#define USERHOOK_SUPERSLOWLOOP userhook_SuperSlowLoop();  // for code to be run at 1hz
+//#define USERHOOK_AUXSWITCH userhook_auxSwitch();  // for code to handle user aux switches
```
[the remark this answers is not in the document]
Comment (SergeyBokhantsev) [reply ArduCopter/APM_Config.h]: Im using #ifdef USERHOOK_AUXSWITCH statement (not checking == ENABLE). It is how all another user hooks defined above. Do you really want me to remake that?

[the remark this answers is not in the document]
Comment (khancyr) [reply ArduCopter/APM_Config.h]: It is not a problem to use ENABLE, it just define USERHOOK_AUXSWITCH to 1 and then you will pass all #ifdef. Maybe change the #ifdef to #if will be better, but as it is compile time only , it shouldn't matter.
Other functions are using define to call the userhook on the scheduler and we don't want that here, that is why you don't need to define to userhook_auxSwitch() function.

File: ArduCopter/UserCode.cpp [hunk 2]
```diff
@@ -42,3 +42,20 @@ void Copter::userhook_SuperSlowLoop()
...
+void Copter::userhook_auxSwitch2(uint8_t ch_flag)
+{
+    // put your aux switch #2 handler here (CHx_OPT = 48)
+}
+
+void Copter::userhook_auxSwitch3(uint8_t ch_flag)
+{
+    // put your aux switch #3 handler here (CHx_OPT = 49)
+}
+#endif
```
[the remark this answers is not in the document]
Comment (SergeyBokhantsev) [reply ArduCopter/UserCode.cpp]: fixed

File: ArduCopter/APM_Config.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (SergeyBokhantsev) [reply ArduCopter/APM_Config.h]: fixed

**Comment by SergeyBokhantsev** [comment]
@khancyr added fixes on your comments

**Comment by khancyr** [comment]
hahah nope ! 
Read again ! You change the title on gihub Pull Request, but not the commit message ! ;-P

Here is some help : http://ardupilot.org/dev/docs/git-interactive-rebase.html
You need to squash you two commits and change the message.

**Comment by SergeyBokhantsev** [comment]
@khancyr got it, done.

File: ArduCopter/UserCode.cpp [hunk 3]
```diff
@@ -42,3 +42,21 @@ void Copter::userhook_SuperSlowLoop()
...
+{
+    // put your aux switch #2 handler here (CHx_OPT = 48)
+}
+
+void Copter::userhook_auxSwitch3(uint8_t ch_flag)
+{
+    // put your aux switch #3 handler here (CHx_OPT = 49)
+}
+
+#endif
```
Comment (khancyr) [inline ArduCopter/UserCode.cpp]: miss new line at the end of file

Comment (SergeyBokhantsev) [reply ArduCopter/UserCode.cpp]: @khancyr ups, I never care about new lines at the EOF before. I believe I fixed it now.

**Comment by rmackay9** [comment]
OK, merged thanks!

**Pull request closed without merging**